### DISCUSSION TITLE
Schematic question

### DISCUSSION THREAD

**Discussion by bmwhitetx** [Dilbert66/esphome-vistaECP#117] [General]
The Readme provides 3 schematics labelled 1, 2, 3. On Schematic 1, Green  routes through a resistor divider and lands on SCK. On Schematic 2, Green  routes through a resistor divider but lands on **CS**.

Based on the three schematics and protocol used seems like Schematic 2 is in error and the line going to CS should go to SCK. CS is used in SPI but it looks like you use I2C. Drawing is readme_material/master_noopto.png

Great Project - still deciding what to build, I like the alternative by aselle since I'll likely order a pcb from JLC. Where did you land on including his schematic (https://github.com/Dilbert66/esphome-vistaECP/issues/49)?

**Comment by rcmurphy** [comment]
This isn’t I2C, it’s a proprietary protocol, so the pin doesn’t matter.On Nov 20, 2023, at 10:09, bmwhitetx ***@***.***> wrote:﻿
The Readme provides 3 schematics labelled 1, 2, 3. On Schematic 1, Green  routes through a resistor divider and lands on SCK. On Schematic 2, Green  routes through a resistor divider but lands on CS.
Based on the three schematics and protocol used seems like Schematic 2 is in error and the line going to CS should go to SCK. CS is used in SPI but it looks like you use I2C. Drawing is readme_material/master_noopto.png
Great Project - still deciding what to build, I like the alternative by aselle since I'll likely order a pcb from JLC. Where did you land on including his schematic (#49)?

**Reply by bmwhitetx** [reply]
Thanks for clarifying, I did read later that it's a 4800 baud communication and that particular line is for Monitor Green. So you're correct in the pin doesn't matter. 

But I think the schematic is still wrong in showing the Green Monitor connecting to physical pin 7 (D8) instead of D5:
1) The note included with schematic 2 says to connect the Monitor Green to GPIO14 (D5)
2) The default YAML file has these definitions:
  ##esp8266
  #rxpin: "5" #GPIO pin to use for data receive (yellow line) 
  #txpin: "4" #GPIO pin to use for data transmit (green line)
  #monitorpin: "14" #GPIO pin to use for monitoring module traffic such as RF or Expanders . Set to -1 to disable

No issue with using ESP32. Both Schematic Note and YAML say wire to GPIO18. So only an issue if a newbie is using a ESP8266 (Wemos D1) and uses the schematic w/o looking at notes or YAML.

**Comment by Dilbert66** [comment]
Yes, it's not correct. Not on my toplist of things to change right now. I'll get to it at some point..

**Comment by Dilbert66** [comment]
Personally I prefer the simple resistor divider shematic.  You do not really gain anything by adding extra complexity to the circuit. I've been using it for years. I've provided the other options for those that want it.  The optocoupled versions have their own issues as they add load to the system.  Aselle's schematic alleviates that but at added complexity and cost. Of course with a board, it's much easier to put together.  I have not tested the various types side by side so can't comment on Aselle's design per say. It is well done though.

**Reply by bmwhitetx** [reply]
Perfect thank you. The more I read I agree that circuit 1 seems like the way to go. Thanks again for sharing all this with the community.

**Comment by dtallard** [comment]
The recommended diagram #1 does not use a USB power supply but it is suggested and recommended that it is better to use one. Diagram #3 shows using an external USB and isolated grounds from the panel and USB powered ESP device but it is the 'least recommended" version. I'm fine with either version but this adds confusion as to which one to build. Does #3 with 3 4N35 Optocouplers perform poorly? Just looking for clarification prior to building. Thanks in advance

**Reply by Dilbert66** [reply]
The easiest and most reliable is the resistor divider version powered by usb.  Using 4n35's for the input side provides a slightly degraded signal quality and increased load on the panel bus but offers some isolation from the esp side. I put it out there since some people wanted it but in my own testing it is sub par.   If you don't have an external power plug available for the usb brick then you can use a buck converter hooked to  the panel and power the esp that way.  It's all up to you.

**Comment by dtallard** [comment]
Thanks for the quick response, I did buy a buck converter and I only have 1 keypad and a 4204 only using 1 relay as a sense for alarm on/off after abandoning my x10 from my vista15. My load is minimal so powering from the panel using a buck seems the best. I plan on adding the virtual keypad to my central dash in my hallway. Thanks again!

⁣Get BlueMail for Android ​

**Comment by dtallard** [comment]
My 8266 has a usbC connector to power it. Can I use that to power the ESP
and use diagram 1 minus the U3 piece to pin9? It would seem to tie the
grounds of the panel and the USB brick together or do I also eliminate
pin10 GND and pin 9 power? Sorry for confusion, just don't want to blow
anything up.

**Comment by Dilbert66** [comment]
You still need all the grounds.  Using the usb power just removes the circuit in the dotted square.

**Comment by dtallard** [comment]
I made the bd using the first schematic and everything measures ok and so far no smoke but haven't hooked it to the panel yet as I want to get it situated in a case with the ESP8266(ESP-12E). I edited the VistaAlarm.yaml to hopefully match my vista15p panel eliminating the extra partitions  that don't exist on a 15. I got the the project to build and upload but the only entity I see is firmware. Does it need to be connected to the panel to get all the other entities like the binary_sensor? (text_sensor , etc.) I do see an ESPHome device with a FW entity but that's all. I was expecting to see them but say 'unavailable' until hooked up. Thanks in advance 

**Update: Nevermind!!** I have both ESP32 and 8266 devices and initially tried to use the 8266 which I got to compile and upload giving me just the 1 FW entity. It compiled but I did have RTC compile warnings that didn't seem to kill the compile so I ignored them. After 2 days I decided to try the ESP32 using the same method of merging the 'initial esphome yaml" with the VistaAlarm yaml edited to my panel specifics. First of all, no RTC compile warnings and it compiled,linked,uploaded and booted and then in the rolling logs I saw errors that looked like it was trying to talk to the panel "Is it connected to the panel?" messages. Then I sucked it into HA and all the entities are there. Now it'ts  time to actually hook it up. I also have to modify my DHCP reservation to the new MAC of the ESP32.
Thanks for a great project to follow!

**Reply by Dilbert66** [reply]
Good stuff.  I highly recommed an esp32 for the project over the esp8266.   Also, i would recommend using platform: esp-idf instead of arduino for a faster compile as well as the added benefit of more streamlined code.

**Comment by dtallard** [comment]
Initially I flashed using chrome with it connected to my laptop using the
empty ESPHome device choice which then allows for OTA and HA sucked it
right in. I then took that very basic yaml file and merged it with your
VistaAlarm.yaml that was customized to my panel
This is what esphome put in and I used:

esp32:
  variant: esp32
  framework:
    type: esp-idf

I should have everything finished today becasue after looking at my first
circuit bd,  it was really ugly and I had wired it to the esp via a
breakout bd so it was 2 bds. My new one is much neater and I used a bigger
perfbd and added some headers to plug in the esp and the single 4n35 plus I
made the cable to the panel 2 halves so I can unplug from the panel with
out disturbing the panel itself at all and now I will be able to gaurantee
no interaction with the panel if I need to do OTA or if there is an issue I
can easily move it to the bench.
Thanks again, Dean
